Sources/Hummingbird/Middleware/MiddlewareModule/MiddlewareStack.swift

```diff
@@ -34,6 +36,47 @@ public struct _Middleware2<M0: MiddlewareProtocol, M1: MiddlewareProtocol>: Midd
     }
 }
 
+extension _Middleware2: RouterMiddleware where M0.Input == Request, M0.Output == Response {}
+
+// MARK: - ConditionalMiddleware
+
+public enum _ConditionalMiddleware<M0: MiddlewareProtocol, M1: MiddlewareProtocol>: MiddlewareProtocol where M0.Input == M1.Input, M0.Context == M1.Context, M0.Output == M1.Output {
```

Comment:
- [ ] With the addition of `_ConditionalMiddleware` and `_EmptyMiddleware`, this file is getting a bit bulky. It might be worthwhile moving all these "public private" middleware unto their own files inside a subfolder.